### ISSUE TITLE
RUF007 - False positive when `zip()` contains more than 2 arguments

### ISSUE THREAD

**Issue by MaxWinterstein** [charliermarsh/ruff#3651]
RUF007 should only hit when `zip()` is called with no more than 2 arguments, as `pairwise()` does not support it. 

Introduced with #3501 

See
- https://docs.python.org/3/library/itertools.html#itertools.pairwise
- https://docs.python.org/3/library/functions.html#zip

`pyproject.toml`
```toml
[tool.ruff]
target-version = "py311"
``` 

`test.py`
```python
foo = [1, 2, 3, 4]

bar = zip(foo[:-1], foo[1:], foo, strict=True)
```

`ruff test.py`
```ruby
test.py:5:7: RUF007 Prefer `itertools.pairwise()` over `zip()` when iterating over successive pairs
Found 1 error.
```

`ruff --version`
```bash
ruff 0.0.257
```

**Comment by charliermarsh** [comment]
Thanks!

**Comment by JonathanPlasse** [comment]
I would like to work on it.
